mmaction/models/backbones/uniformer.py

```diff
@@ -0,0 +1,663 @@
+            Defaults to 224.
+        patch_size (int): Number of patch size.
+            Defaults to 16.
+        in_chans (int): Number of input features.
+            Defaults to 3.
+        embed_dim (int): Number of output features.
+            Defaults to 768.
+        init_cfg (dict, optional): Config dict for initialization.
+            Defaults to None.
+    """
+
+    def __init__(
+        self,
+        img_size=224,
+        patch_size=16,
+        in_chans=3,
+        embed_dim=768,
+        init_cfg: Optional[dict] = None,
+    ) -> None:
+        super().__init__(init_cfg=init_cfg)
+
+        img_size = to_2tuple(img_size)
+        patch_size = to_2tuple(patch_size)
+        num_patches = (img_size[1] // patch_size[1]) * (
+            img_size[0] // patch_size[0])
+        self.img_size = img_size
+        self.patch_size = patch_size
+        self.num_patches = num_patches
+        self.norm = nn.LayerNorm(embed_dim)
+        self.proj = conv_3xnxn(
+            in_chans,
+            embed_dim,
+            kernel_size=patch_size[0],
+            stride=patch_size[0])
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.proj(x)
+        B, _, T, H, W = x.shape
+        x = x.flatten(2).transpose(1, 2)
+        x = self.norm(x)
+        x = x.reshape(B, T, H, W, -1).permute(0, 4, 1, 2, 3).contiguous()
+        return x
+
+
+class PatchEmbed(BaseModule):
+    """Image to Patch Embedding.
+
+    Args:
+        img_size (int): Number of input size.
+            Defaults to 224.
+        patch_size (int): Number of patch size.
+            Defaults to 16.
+        in_chans (int): Number of input features.
+            Defaults to 3.
+        embed_dim (int): Number of output features.
+            Defaults to 768.
+        init_cfg (dict, optional): Config dict for initialization.
+            Defaults to None.
+    """
+
+    def __init__(
+        self,
+        img_size=224,
+        patch_size=16,
+        in_chans=3,
+        embed_dim=768,
+        init_cfg: Optional[dict] = None,
+    ) -> None:
+        super().__init__(init_cfg=init_cfg)
+
+        img_size = to_2tuple(img_size)
+        patch_size = to_2tuple(patch_size)
+        num_patches = (img_size[1] // patch_size[1]) * (
+            img_size[0] // patch_size[0])
+        self.img_size = img_size
+        self.patch_size = patch_size
+        self.num_patches = num_patches
+        self.norm = nn.LayerNorm(embed_dim)
+        self.proj = conv_1xnxn(
+            in_chans,
+            embed_dim,
+            kernel_size=patch_size[0],
+            stride=patch_size[0])
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.proj(x)
+        B, _, T, H, W = x.shape
+        x = x.flatten(2).transpose(1, 2)
+        x = self.norm(x)
+        x = x.reshape(B, T, H, W, -1).permute(0, 4, 1, 2, 3).contiguous()
+        return x
+
+
+@MODELS.register_module()
+class UniFormer(BaseModule):
+    """UniFormer.
```

Comment:
please add paper link for Uniformer, take [swin](https://github.com/open-mmlab/mmaction2/blob/dev-1.x/mmaction/models/backbones/swin.py#L707) as a reference.

Reply:
Done